io.openems.edge.common/src/io/openems/edge/common/channel/Channel.java

```diff
@@ -112,9 +112,26 @@ public default void setNextValue(Object value) {
 	 * Add an onSetNextValue callback. It is called, after a new NextValue was set.
 	 * Note that usually you should prefer the onUpdate() callback.
 	 * 
+	 * <p>
+	 * 
+	 * Remember to remove the callback using
+	 * {@link #removeOnSetNextValueCallback(Consumer)} once it is not needed
+	 * anymore, e.g. on deactivate().
+	 * 
 	 * @see #onUpdate
+	 * 
+	 * @param callback the callback {@link Consumer}
+	 * @return the callback to enable fluent programming
+	 */
+	public Consumer<Value<T>> onSetNextValue(Consumer<Value<T>> callback);
+
+	/**
+	 * Removes an onSetNextValue callback.
+	 * 
+	 * @see #onSetNextValue(Consumer)
+	 * @param callback the callback {@link Consumer}
 	 */
-	public void onSetNextValue(Consumer<Value<T>> callback);
+	public void removeOnSetNextValueCallback(Consumer<?> callback);
```

Comment:
The 'add' method is `onSetNextValue()`. I agree for the change, but if I would have renamed this as well, I would have touched a hundred other files which I tried to avoid. Should be a separate pull-request (or we keep it for legacy, because `onSetNextValue()` is very good readable)